### PR TITLE
Add tests for color_filtered.0.dart example.

### PR DESCRIPTION
**Pull request by ksokolovskyi**
This PR contributes to https://github.com/flutter/flutter/issues/130459

### Description
- Adds tests for `examples/api/lib/widgets/color_filter/color_filtered.0.dart`

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

### CODE REVIEW

File: examples/api/test/widgets/color_filter/color_filtered.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  // The app being tested loads images via HTTP which the test
+  // framework defeats by default.
+  setUpAll(() {
+    HttpOverrides.global = null;
+  });
+
+  testWidgets('Displays two images', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline examples/api/test/widgets/color_filter/color_filtered.0_test.dart:20]: These 3 tests can be just one.

"'Displays two images" test - you're already finding first and last images the implies are only two images.

"'Applies red color filter with modulate blend mode to the first image"  & "Applies grey color filter with saturation blend mode to the second image" - These two short and too verbose. We can add single test with "Color filter is applied to the images" and explain the difference in the test comment instead.

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: examples/api/test/widgets/color_filter/color_filtered.0_test.dart (hunk 1 above)
Comment (ksokolovskyi) [reply examples/api/test/widgets/color_filter/color_filtered.0_test.dart]: @TahaTesser thanks for the review and great suggestion. I applied it in the latest commit.

File: examples/api/test/widgets/color_filter/color_filtered.0_test.dart [hunk 2]
```diff
@@ -17,50 +17,37 @@ void main() {
...
-
-  testWidgets('Applies red color filter with modulate blend mode to the first image', (WidgetTester tester) async {
-    await tester.pumpWidget(
-      const example.ColorFilteredExampleApp(),
-    );
-
-    await tester.pumpAndSettle();
 
-    final RenderObject renderObject = tester.firstRenderObject(
+    final RenderObject renderObject1 = tester.firstRenderObject(
```
Comment (TahaTesser) [inline examples/api/test/widgets/color_filter/color_filtered.0_test.dart:30]: We can remove final and reassign the value, this also ensures that we have updated expected values.

File: examples/api/test/widgets/color_filter/color_filtered.0_test.dart [hunk 3]
```diff
@@ -17,50 +17,37 @@ void main() {
...
       find.byType(ColorFiltered).first,
     );
-    final ColorFilterLayer colorFilterLayer =
-        renderObject.debugLayer! as ColorFilterLayer;
+    final ColorFilterLayer colorFilterLayer1 =
```
Comment (TahaTesser) [inline examples/api/test/widgets/color_filter/color_filtered.0_test.dart:33]: We can remove final and reassign the value, this also ensures that we have updated expected values.

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: examples/api/test/widgets/color_filter/color_filtered.0_test.dart [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    await tester.pumpWidget(
+      const example.ColorFilteredExampleApp(),
+    );
+
+    await tester.pumpAndSettle();
+
+    // Verify that two images are displayed.
+    expect(find.byType(Image), findsNWidgets(2));
+
+    final RenderObject renderObject1 = tester.firstRenderObject(
```
Comment (TahaTesser) [inline examples/api/test/widgets/color_filter/color_filtered.0_test.dart:30]:
```suggestion
    RenderObject renderObject = tester.firstRenderObject(
```

File: examples/api/test/widgets/color_filter/color_filtered.0_test.dart [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    final ColorFilterLayer colorFilterLayer1 =
+        renderObject1.debugLayer! as ColorFilterLayer;
+
+    // Verify that red colored filter with modulate blend mode is applied to the first image.
+    expect(
+      colorFilterLayer1.colorFilter,
+      equals(const ColorFilter.mode(Colors.red, BlendMode.modulate)),
+    );
+
+    final RenderObject renderObject2 = tester.firstRenderObject(
```
Comment (TahaTesser) [inline examples/api/test/widgets/color_filter/color_filtered.0_test.dart:42]:
```suggestion
    renderObject = tester.firstRenderObject(
```

Earlier we found render object for the first image, here we assign the variable to expect a different value.

**Review by TahaTesser — CHANGES_REQUESTED** [review]

**Review by TahaTesser — APPROVED** [review]
LGTM

**Review by bleroux — APPROVED** [review]
LGTM!

**Comment by ksokolovskyi** [comment]
@TahaTesser @bleroux thanks for the review!

**Pull request merged**